src/plugins/intel_npu/src/plugin/src/compiled_model.cpp

```diff
@@ -150,12 +153,12 @@ void CompiledModel::export_model(std::ostream& stream) const {
     std::stringstream str;
     str << "Blob size: " << blob.size() << ", hash: " << std::hex << hash(blob);
     _logger.info(str.str().c_str());
+
     // If graphHandle is not a nullptr it means there is still an instance of the blob maintained inside the driver and
     // we can release the copy of the blob here to reduce memory consumption.
-    if (_networkPtr->metadata.graphHandle != nullptr) {
-        auto& blobFilled = const_cast<NetworkDescription*>(_networkPtr.get())->compiledNetwork;
-        blobFilled.clear();
-        blobFilled.shrink_to_fit();
+    auto cidGraph = std::dynamic_pointer_cast<CiDGraph>(_graph);
+    if (cidGraph) {
+        cidGraph->releaseCompiledNetwork();
```

Comment:
To support this, we can not pass std::shared_ptr<const IGraph> everywhere, if we will change something, we shall not mark as const